app/containers/TrackDetailsContainer/index.js

```diff
@@ -0,0 +1,176 @@
+import React, { useEffect, useState, useRef, memo } from 'react';
+import { isEmpty } from 'lodash';
+import PropTypes from 'prop-types';
+import T from '@components/T';
+import If from '@components/If';
+import styled from 'styled-components';
+import { Card, Button, Image } from 'antd';
+import { selectTrackDetailsData, selectTrackDetailsError, selectTrackId } from './selectors';
+import { trackDetailsCreators } from './reducer';
+import { connect } from 'react-redux';
+import trackDetailsSaga from './saga';
+import { injectIntl, FormattedMessage } from 'react-intl';
+import { injectSaga } from 'redux-injectors';
+import { compose } from 'redux';
+import { createStructuredSelector } from 'reselect';
+import { useParams } from 'react-router-dom';
+
+const CustomCard = styled(Card)`
+  && {
+    margin: 20px 0;
+    max-width: ${(props) => props.maxwidth};
+    color: ${(props) => props.color};
+    ${(props) => props.color && `color: ${props.color}`};
+  }
+`;
+
+const Container = styled.div`
+  && {
+    margin: 1.9rem auto;
+    padding: 20px;
+    height: max-content;
+    width: 40rem;
+    box-shadow: 0 5px 20px -5px rgb(191 192 192);
+    border-radius: 0.8rem;
+  }
+`;
+
+const ImageWrapper = styled.div`
+  display: flex;
+  gap: 1.4rem;
+`;
+
+const TrackDetailsSection = styled.div``;
+
+const TrackImage = styled(Image)`
+  width: ${(props) => props.width}px;
+`;
+
+const PlayBtn = styled(Button)`
+  type: ${(props) => props.type};
+`;
+
+export function TrackDetailsContainer({ dispatchTrackDetails, trackDetailsData, trackDetailsError, loading }) {
```

Comment:
Isn't `loading` undefined? It's not getting `loading` state from redux and also not defined in reducer. Handle the loading state from reducer in ItunesContainer also.

Reply:
yeah, updated the loading state!